zebra-chain/src/primitives/address.rs

```diff
@@ -57,7 +59,7 @@ impl From<Network> for zcash_address::Network {
     fn from(network: Network) -> Self {
         match network {
             Network::Mainnet => zcash_address::Network::Main,
-            Network::Testnet => zcash_address::Network::Test,
+            Network::Testnet(_) => zcash_address::Network::Test,
```

Comment:
This is incorrect, this conversion is only valid if `network == Network::new_testnet()`.

Otherwise, it is an unsupported network.